libobs/util/pipe-posix.c

```diff
@@ -16,34 +16,112 @@
 
 #include <stdio.h>
 #include <sys/wait.h>
+#include <unistd.h>
+#include <errno.h>
+#include <spawn.h>
 
 #include "bmem.h"
 #include "pipe.h"
 
 struct os_process_pipe {
 	bool read_pipe;
+	int pid;
 	FILE *file;
+	FILE *err_file;
 };
 
 os_process_pipe_t *os_process_pipe_create(const char *cmd_line,
 					  const char *type)
 {
-	struct os_process_pipe pipe = {0};
+	struct os_process_pipe process_pipe = {0};
 	struct os_process_pipe *out;
+	posix_spawn_file_actions_t file_actions;
 
 	if (!cmd_line || !type) {
 		return NULL;
 	}
 
-	pipe.file = popen(cmd_line, type);
-	pipe.read_pipe = *type == 'r';
+	process_pipe.read_pipe = *type == 'r';
+
+	int mainfds[2], errfds[2];
```

Comment:
Code style: Use two separate lines for declaration. I guess the arrays are initialised with zeroes by default, so additional initialisation is not necessary.